src/posix/platform/hdlc_interface.cpp

```diff
@@ -235,19 +235,16 @@ otError HdlcInterface::Write(const uint8_t *aFrame, uint16_t aLength)
     {
         ssize_t rval = write(mSockFd, aFrame, aLength);
 
-        if (rval > 0)
+        if (rval >= 0)
         {
             aLength -= static_cast<uint16_t>(rval);
             aFrame += static_cast<uint16_t>(rval);
-            continue;
+            SuccessOrExit(error = WaitForWritable());
```

Comment:
Looks good as is.
One suggestion: Can we add a check to see if `aLength` is now zero after the `-=` and `break` from while loop to avoid an extra call to `WaitForWritable()` when all bytes are sent.

Reply:
updated.